Squeezer/src/main/java/uk/org/ngo/squeezer/framework/BaseActivity.java

```diff
@@ -426,18 +428,19 @@ public void showDisplayMessage(DisplayMessage display) {
             showMe = false;
         }
 
-        if (showMe) {
-            if (!(icon.getVisibility() == View.VISIBLE &&text.getVisibility() == View.VISIBLE)) {
-                layout.findViewById(R.id.divider).setVisibility(View.GONE);
-            }
-            int duration = (display.duration >=0 && display.duration <= 3000 ? Toast.LENGTH_SHORT : Toast.LENGTH_LONG);
-            Toast toast = new Toast(getApplicationContext());
-            //TODO handle duration == -1 => LENGTH.INDEFINITE and custom (server side) duration,
-            // once we have material design and BaseTransientBottomBar
-            toast.setDuration(duration);
-            toast.setView(layout);
-            toast.show();
+        if (!showMe) {
+            return;
+        }
+        if (lastShownToast != null) {
+            lastShownToast.cancel();
```

Comment:
This look suspicious, e.g., might cause illegal state exception when the Toast is not showing.
Do you have any experience with this construct?

Reply:
Well, I double checked in the [Android sources](https://android.googlesource.com/platform/frameworks/base/+/refs/heads/android14-release/core/java/android/widget/Toast.java) and there's no obvious reason why `IllegalStateException` should potentially be thrown. Do you have any indication for this potentially being the case?
In either case, the problem of 'successive display messages are not shown properly' definitely is a real one.

Reply:
This is just a precaution, because I regularly receive crashes from production on Google Play Console, which I haven't seen in test and not even for any beta users. Sometimes those crashes relates to vendors which have made changes to the Android source code, or to specific Android versions, where the semantics might differ.
But I think this should be fine.